File: back-end/src/main/java/com/tfg/back/model/dtos/LabResults/LabResultDtoGet.java
package com.tfg.back.model.dtos.labResults;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class LabResultDtoGet {

    private Long id;

    private String testName;

    private String resultValue;

    private String unit;

    private String referenceRange;

    private LocalDate resultDate;

    private String notes;

    private String patientFullName;

    private String doctorFullName;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private Long version;

    private String status;

}

File: front-end/src/app/models/lab-result-dto-get.ts
export interface LabResultDtoGet {
  id: number;
  testName: string;
  resultValue: string;
  unit: string;
  referenceRange: string;
  resultDate: string; 
  notes: string;
  patientFullName: string;
  doctorFullName: string;
  createdAt: string; 
  updatedAt: string;
  version: number;
  status: string;
}